packages/ns-api/README.md

```diff
@@ -5343,6 +5343,124 @@ Response example:
 {"result": "success"}
 ```
 
+## ns.threatshield
+
+Manage banip configuration.
+
+### list-blocklist
+
+List current blocklist:
+```
+api-cli ns.threatshield list-blocklist
+```
+
+Response example:
+```json
+{
+  "data": [
+    {
+      "name": "yoroimallvl1",
+      "type": "enterprise",
+      "enabled": false,
+      "confidence": 10
+    },
+    {
+      "name": "yoroimallvl2",
+      "type": "enterprise",
+      "enabled": false,
+      "confidence": 8
+    }
+  ]
+}
+```
+
+Fields:
+- type can be `enterprise`, `community` or `warning`,  the warning type is when the list is present in UCI configuration but it's not a supported feed
+- confidence can be `-1` if the value is not available
+
+
+### list-settings
+
+Show current banip settings:
+```
+api-cli ns.threatshield list-settings
+```
+
+Response example:
+```json
+{"data": {"enabled": true}}
+```
+
+### edit-settings
+
+Configure banip settings
+```
+api-cli ns.threatshield edit-settings --data '{"enabled": true}'
+```
+
+Response example:
+```json
+{"message": "success"}
+```
+
+### list-allowed
+
+List addresses always allowed:
+```
+api-cli ns.threatshield list-allowed
+```
+
+Response example:
+```json
+```
+
+### add-allowed
+
+Add an address which is always allowed:
+```
+api-cli ns.threatshield add-allowed --data '{"address": "1.2.3.4", "description": "my allow1"}'
+```
+
+The `address` field can be an IPv4/IPv6, a CIDR, a MAC or host name
```

Comment:
Hostnames and IP ranges are not listed here by choice or by mistake?

Reply:
host name is already listed, IP ranges are not supported